### DISCUSSION TITLE
自動Debug AI

### DISCUSSION THREAD

**Discussion by ayakasanma** [1101-GS3073-C/Python-Community-GS3073-C#190] [General]
看來在不遠的未來我們已經可以靠AI幫我們解決惱人的小臭蟲，不僅省時也讓寫程式的心情愉快很多！
https://www.ithome.com.tw/news/147404

**Comment by Timmy110403540** [comment]
以後可以多多利用這個程式，有時候看錯誤在哪會看到頭很痛XD

**Comment by wh20502** [comment]
其實我滿好奇它能夠偵測的限制是什麼，什麼樣的BUG會被修正，感覺明年可以拭目以待。

**Comment by Shiba010** [comment]
這太厲害了，debug往往是工程師最頭痛的。

**Reply by ayakasanma** [reply]
確實！

**Comment by ghost** [comment]
能夠debug的AI能不能把自己的BUG給de出來呢XD

**Comment by Lynn12378** [comment]
有時候程式碼看久了真的會打錯很多東西，但是不知道從何改起，這個功能感覺很讚

**Comment by qeeeee39** [comment]
期待最後成品，也期待自己debug能力可以跟這個ai一樣強。

**Comment by 110409538** [comment]
不管是對程式的初學者或是工程師而言，我覺得這個工具非常方便。

**Comment by Yunhans** [comment]
我寫c常常少打分號很需要

**Comment by LanningYang** [comment]
能提醒部份的問題點，解決仍須靠工程師邏輯

**Comment by lx3211** [comment]
其實寫程式最花時間就是在處理那些臭bug！！！
用AI殺蟲劑的時代到了

**Comment by boson13579** [comment]
能幫忙debug的AI幫工程師減少了很多時間。工程師往往花費過多時間和精力只為了抓小小一兩處十分細小的錯誤，因此而耽誤了專案時間，拖垮專案進度。

**Comment by larry40305** [comment]
結合自動寫程式的AI，工程師就都不用自己寫程式了

**Comment by HsuChen88** [comment]
對不熟悉的人來說debug似乎是寫程式最花時間的階段了吧，如果有AI能輔助的話，會少走很多冤枉路。

**Comment by ghost** [comment]
寫一大串程式最惱人的大概就是出錯之後，需要花時間抽絲剝繭地找出錯誤的地方，如果有這個，那麼工程師就能減少非常多的時間DEBUG，是非常有用的技術

**Comment by Yilin20021016** [comment]
那麼的話在打程式可以更加的快了 因為不再要debug整天

**Comment by Bonus502530** [comment]
能自動debug對有在寫程式的人幫助蠻大的，不過雖然它可以自動debug，但是我想到之前在練CPE的考古題時，有一題我debug 1個多小時才找出錯誤，結果是邊界值的問題，那題應開要開long long，不過我只開 int用 uva online judge就是過不了，我好奇像這樣的bug可以de的到嗎 ?

**Comment by xCk27x** [comment]
我覺得這項技術很難真正成功ㄟ 
- 要讓AI知道正確的程式運作起來會是甚麼狀況
- 需要考慮到非常多複雜 不具體的邏輯思維
- 要能涵蓋多個語言又會是一項曠日廢時的工作==

但我很希望這項技術能夠融入現有IDE的debug工具中
光是能提醒我哪邊可以修正就已經夠好用了~

**Comment by bww0401** [comment]
不得不說，有時候程式幾百行要抓蟲真的是超級燥，花了很多時間只為了抓蟲真的不值...，如果未來這個程式可以用，我一定要好好用它來抓臭蟲，還是說這個程式可以結合自動撰寫程式XD
